src/plugins/ThumbnailGenerator/index.js

```diff
@@ -104,13 +104,12 @@ module.exports = class ThumbnailGenerator extends Plugin {
 
     image = this.protect(image)
 
-    var steps = Math.ceil(Math.log2(image.width / targetWidth))
+    var steps = Math.ceil(Math.abs(Math.log2(image.width / targetWidth)))
```

Comment:
I'm not sure if `Math.abs` is the best way to do it, for very small images the Math.abs call might return a positive number > 1 which would cause the step scaling to first size it up to N times the target size and then resize it back down. Another way could be to check if `steps < 1` and then set it to 1?